File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamQuaTang/SanPhamQuaTangColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.SanPhamQuaTang")]
[BasedOnRow(typeof(SanPhamQuaTangRow), CheckNames = true)]
public class SanPhamQuaTangColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string SanPhamTenSp { get; set; }
    public string QuaTangTenQua { get; set; }
    public int ThuTu { get; set; }
    public bool IsDefault { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamQuaTangColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SanPhamQuaTangRow } from "./SanPhamQuaTangRow";

export interface SanPhamQuaTangColumns {
    Id: Column<SanPhamQuaTangRow>;
    SanPhamTenSp: Column<SanPhamQuaTangRow>;
    QuaTangTenQua: Column<SanPhamQuaTangRow>;
    ThuTu: Column<SanPhamQuaTangRow>;
    IsDefault: Column<SanPhamQuaTangRow>;
}

export class SanPhamQuaTangColumns extends ColumnsBase<SanPhamQuaTangRow> {
    static readonly columnsKey = 'SanPhamDB.SanPhamQuaTang';
    static readonly Fields = fieldsProxy<SanPhamQuaTangColumns>();
}